courseware/api.py

```diff
@@ -226,6 +227,66 @@ def _create_tokens_and_update_auth(auth, params):
     return auth
 
 
+def repair_faulty_edx_user(user):
+    """
+    Loops through all Users that are incorrectly configured in edX and attempts to get
+    them in the correct state.
+
+    Args:
+        user (User): User to repair
+        platform (str): The courseware platform
+
+    Returns:
+        (bool, bool): Flags indicating whether a new edX user was created and whether a new
+                edX auth token was created.
+    """
+    created_user, created_auth_token = False, False
+    if (
+        find_object_with_matching_attr(
+            user.courseware_users.all(), "platform", value=PLATFORM_EDX
+        )
+        is None
+    ):
+        create_edx_user(user)
+        created_user = True
+    if not hasattr(user, "openedx_api_auth"):
+        create_edx_auth_token(user)
+        created_auth_token = True
+    return created_user, created_auth_token
+
+
+def repair_faulty_courseware_users():
+    """
+    Loops through all Users that are incorrectly configured with the courseware and attempts to get
+    them in the correct state.
+
+    Returns:
+        list of User: Users that were successfully repaired
+    """
+    repaired_users = []
+    for user in User.faulty_courseware_users.all():
```

Comment:
You mentioned potential alarm fatigue, maybe it's worth filtering out accounts created in the 2-3 minutes or so?